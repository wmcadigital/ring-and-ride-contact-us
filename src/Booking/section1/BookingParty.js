import { useEffect } from "react";
import PropTypes from "prop-types";
import { useFormState } from "react-final-form";

import FormSection from "../../common/FormSection";
import RadioGroup from "../../common/RadioGroup";
import Question from "../../common/Question";
import RadioButton from "../../common/RadioButton";
import FieldError from "../../common/FieldError";
import { required } from "../../common/validation";

const BookingParty = ({ setBookingParty }) => {
  const stateApi = useFormState();
  const formValues = stateApi.values;

  const error = stateApi.submitFailed ? stateApi.errors?.bookingParty : null;

  useEffect(() => {
    if (formValues.bookingParty) {
      setBookingParty(formValues.bookingParty);
    }
  }, [formValues.bookingParty]);

  return (
    <FormSection>
      <Question text="Who are you making the booking for?" />
      <RadioGroup error={error}>
        <FieldError text={error} />
        <RadioButton
          key={1}
          label="Myself"
          validation={required}
          value="mySelf"
          fieldName="bookingParty"
        />
        <RadioButton
          key={2}
          label="On behalf of someone else"
          validation={required}
          value="behalfSomeone"
          fieldName="bookingParty"
        />
        <RadioButton
          key={3}
          label="On behalf of a group"
          validation={required}
          value="behalfGroup"
          fieldName="bookingParty"
        />
      </RadioGroup>
    </FormSection>
  );
};

BookingParty.propTypes = {
  setBookingParty: PropTypes.func,
};

BookingParty.defaultProps = {
  setBookingParty: () => {},
};

export default BookingParty;
