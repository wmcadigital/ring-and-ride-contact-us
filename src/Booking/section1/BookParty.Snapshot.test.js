import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import BookingParty from "./BookingParty";

jest.mock("react-final-form");

describe("BookingParty - snapshots", () => {
  it("renders as expected with no value selected", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {},
    }));
    const renderer = createRenderer();

    renderer.render(<BookingParty />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders as expected with value selected", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingParty />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {},
      submitFailed: true,
      errors: { bookingParty: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingParty />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
