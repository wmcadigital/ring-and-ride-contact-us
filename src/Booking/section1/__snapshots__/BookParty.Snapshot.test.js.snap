// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingParty - snapshots renders as expected with no value selected 1`] = `
<FormSection>
  <Question
    text="Who are you making the booking for?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="bookingParty"
      label="Myself"
      onClickCallback={[Function]}
      validation={[Function]}
      value="mySelf"
    />
    <RadioButton
      fieldName="bookingParty"
      label="On behalf of someone else"
      onClickCallback={[Function]}
      validation={[Function]}
      value="behalfSomeone"
    />
    <RadioButton
      fieldName="bookingParty"
      label="On behalf of a group"
      onClickCallback={[Function]}
      validation={[Function]}
      value="behalfGroup"
    />
  </RadioGroup>
</FormSection>
`;

exports[`BookingParty - snapshots renders as expected with value selected 1`] = `
<FormSection>
  <Question
    text="Who are you making the booking for?"
  />
  <RadioGroup
    error={null}
  >
    <FieldError
      text={null}
    />
    <RadioButton
      fieldName="bookingParty"
      label="Myself"
      onClickCallback={[Function]}
      validation={[Function]}
      value="mySelf"
    />
    <RadioButton
      fieldName="bookingParty"
      label="On behalf of someone else"
      onClickCallback={[Function]}
      validation={[Function]}
      value="behalfSomeone"
    />
    <RadioButton
      fieldName="bookingParty"
      label="On behalf of a group"
      onClickCallback={[Function]}
      validation={[Function]}
      value="behalfGroup"
    />
  </RadioGroup>
</FormSection>
`;

exports[`BookingParty - snapshots renders with an error 1`] = `
<FormSection>
  <Question
    text="Who are you making the booking for?"
  />
  <RadioGroup
    error="Required"
  >
    <FieldError
      text="Required"
    />
    <RadioButton
      fieldName="bookingParty"
      label="Myself"
      onClickCallback={[Function]}
      validation={[Function]}
      value="mySelf"
    />
    <RadioButton
      fieldName="bookingParty"
      label="On behalf of someone else"
      onClickCallback={[Function]}
      validation={[Function]}
      value="behalfSomeone"
    />
    <RadioButton
      fieldName="bookingParty"
      label="On behalf of a group"
      onClickCallback={[Function]}
      validation={[Function]}
      value="behalfGroup"
    />
  </RadioGroup>
</FormSection>
`;
